test/testcommon/src/main/java/io/pravega/test/common/TestUtils.java

```diff
@@ -85,6 +86,34 @@ public static void await(Supplier<Boolean> condition, int checkFrequencyMillis,
         }
     }
 
+    @SneakyThrows(InterruptedException.class)
+    public static <T> T awaitEvaluateExpr(T expected, Callable<T> eval, int checkFrequencyMillis, long timeoutMillis) throws TimeoutException {
+
+        T result = null;
+        Supplier<Boolean> condition = null;
+        long remainingMillis = timeoutMillis;
+
+        while (condition == null || !condition.get() && remainingMillis > 0) {
+            try {
+                result = eval.call();
+                T finalResult = result;
+                condition = () -> (expected == null && finalResult == null)
+                        || (expected != null && expected.equals(finalResult));
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
```

Comment:
Why this catch? We don't want to necessarily change the exception type.

Reply:
`eval.call` throws this exception, and the way the catch block handles it has been kept similar to the way things were done prior to the code restructuring.